src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -434,13 +437,13 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		return cursorYAbsolute > lastCommandYAbsolute;
 	}
 
-	private _cursorLineLooksLikeWindowsPrompt(): boolean {
+	private _cursorLineLooksLikeWindowsPrompt(): RegExpMatchArray | undefined {
 		const line = this._terminal.buffer.active.getLine(this._terminal.buffer.active.baseY + this._terminal.buffer.active.cursorY);
 		if (!line) {
-			return false;
+			return;
 		}
 		// TODO: fine tune prompt regex to accomodate for unique configurtions.
-		return line.translateToString(true)?.match(/^(PS.+>)|([A-Z]:\\.*>)/) !== null;
+		return line.translateToString(true)?.match(/^(PS.+>)|([A-Z]:\\.*>)/) ?? undefined;
```

Comment:
Could we use a named capture group for this: `/^(?<prompt>(?:PS.+>)|(?:[A-Z]:\\.*>))/`, then return object describing the prompt (`{ prompt: string } | undefined`)? Then dealing with the regex is all encapsulated within this function.

Reply:
Yeah, I like the idea of that, but I think it's simpler to just return that capture group, if any?